Rego/OneDriveConfig.rego

```diff
@@ -92,6 +92,7 @@ tests[{
 DefaultLinkPermissionPolicy[Policy]{
     Policy := input.SPO_tenant_info[_]
     Policy.DefaultLinkPermission == 1
+    Policy.OneDriveSharingCapability == 1
 }
 
```

Comment:
Thank you! I will do some investigation, there are few more json changes due to the change that Ted proposed, some are due to the external sharing slider change to anyoneIt. It is likely i have missed a few that is related to checking user permissions.

Reply:
I have done a comparison between original setting (external sharing slider at new and existing guests)  vs. test setting (external sharing slider at anyone with both view and edit) vs (external sharing slider at anyone with view only) and found out that 
- **OneDriveSharingCapability** changes is actually due to the changes of external sharing. When it is set to anyone it became 2, and when it set to new and existing guests it changes back to 1. So i have **removed** that line. 
- The original **DefaultLinkPermission** are always 1 for all cases so its irrelevant. So i have **removed** that line. 

In addition, you were right **FileAnonymousLinkType** and **FolderAnonymousLinkType** setting is the actual output that policy 2.3 should be looking at.

I _adjusted the rego_ so that it reflects all cases and _changed the requirement_ to "Anyone file and folder permissions SHOULD be limited to View" since the LinkPermission is incorrect. 

to test out my changes, i have tried 5 cases:

- original setting (external sharing slider at new and existing guests)  -- report pass
- test_1 (external sharing slider at anyone with both view and edit)  -- report shows 2.1 & 2.3 failed with error message "Requirement not met: both file and folder are not limited to view for Anyone"
- test_2 (external sharing slider at anyone with view only) -- report shows 2.1 failed
- test_3 (external sharing slider at anyone with files view and edit and folder view only) -- report shows 2.1 & 2.3 failed with error message "Requirement not met: folder are not limited to view for Anyone"
- test_4 (external sharing slider at anyone with files view only and folder view only and edit) -- report shows 2.1 & 2.3 failed with error message "Requirement not met: file are not limited to view for Anyone"